utils/sorting.hh

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <stdexcept>
+#include <vector>
+#include <map>
+#include <seastar/core/future.hh>
+#include "seastar/coroutine/maybe_yield.hh"
+
+namespace utils {
+
+/**
+ * Topological sort a DAG using Kahn's algorithm.
+ *
+ * @param vertices      Contains all vertices of the graph
+ * @param adjecency_map Entry {k, v} means there is an k->v edge in the graph
+ * @return a vector of topologically sorted vertices
+ * @throws std::runtime_error when a graph has any cycle.
+ */
+template<typename T, typename Compare = std::less<T>>
+seastar::future<std::vector<T>> topological_sort(const std::vector<T>& vertices, const std::multimap<T, T, Compare>& adjecency_map) {
+    std::map<T, uint32_t, Compare> ref_count_map; // Contains counters how many edges point (referece) to a vertex
```

Comment:
Why `uint32_t` and not `size_t`? I suppose it doesn't really matter here as we won't operate on such large data, but it is more idiomatic IMO to use `size_t` to represent counts.

Reply:
yes